code/game/objects/machinery/tesla_turret.dm

```diff
@@ -0,0 +1,190 @@
+/obj/item/tesla_turret
+	name = "tesla turret"
+	desc = "A turret that drains plasma of nearby xenomorphs."
+	icon = 'icons/obj/machines/deployable/sentry/tesla.dmi'
+	icon_state = "grounding_rod_open0"
+	max_integrity = 250
+	var/obj/item/cell/battery
+
+/obj/item/tesla_turret/Initialize(mapload)
+	. = ..()
+	var/matrix/M = transform
+	M.Scale(0.75)
+	transform = M
+
+	AddComponent(/datum/component/deployable_item, /obj/machinery/deployable/tesla_turret, 2 SECONDS, 4 SECONDS)
+
+/obj/item/tesla_turret/Destroy()
+	QDEL_NULL(battery)
+	return ..()
+
+/obj/item/tesla_turret/examine(mob/user)
+	. = ..()
+	if(!battery || !in_range(src, user))
+		return
+	. += span_notice("There is \a [battery] inside. [battery.charge]/[battery.maxcharge] left.")
+	. += span_notice("Alt-click to remove it.")
+
+/obj/item/tesla_turret/AltClick(mob/user)
+	. = ..()
+	if(!in_range(src, user))
```

Comment:
why altclick instead of attackby altenrate?

Reply:
Kinda used to some other codebases where right-clicking is possible, but not the norm, will get it changed for yall.